File: mobile/lib/domain/entities/course.dart
import 'package:equatable/equatable.dart';

class Course extends Equatable {
  final String id;
  final String title;
  final String description;
  final String language;
  final String difficulty;
  final String? imageUrl;

  const Course({
    required this.id,
    required this.title,
    required this.description,
    required this.language,
    required this.difficulty,
    this.imageUrl,
  });

  @override
  List<Object?> get props => [id, title, description, language, difficulty, imageUrl];
}

File: backend/src/models/Course.js
const mongoose = require('mongoose');

const courseSchema = new mongoose.Schema({
  title: { type: String, required: true, trim: true },
  description: { type: String, default: '' },
  language: { type: String, default: 'en' },
  difficulty: {
    type: String,
    enum: ['beginner', 'intermediate', 'advanced'],
    default: 'beginner',
  },
  imageUrl: { type: String, default: '' },
}, {
  timestamps: true,
  toJSON: {
    transform(doc, ret) {
      ret.id = ret._id;
      delete ret._id;
      delete ret.__v;
      return ret;
    },
  },
});

module.exports = mongoose.model('Course', courseSchema);
